cmd/otelcol/main.go

```diff
@@ -113,6 +115,31 @@ func main() {
 	}
 }
 
+// required to support --set functionality no longer directly parsed by the core config loader.
+// taken from https://github.com/open-telemetry/opentelemetry-collector/blob/48a2e01652fa679c89259866210473fc0d42ca95/service/flags.go#L39
+type stringArrayValue struct {
```

Comment:
is it worth adding the interface check for flag.Value for this?

Reply:
I think there's a compile time check for line 137 and don't think the stdlib will be updating their type anytime soon.  If we were using this w/ indirection/another interface I would though.